enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+```
+
+The PO manager will be responsible for acting as a proxy between the CVO, and the underlying PO packages during phase 0. This can be done by reconciling an "aggregate" ClusterOperator that reflects the overall state of installing PO packages. Future phases may require PO packages to reconcile OLM's OperatorCondition API, or the directly manage a ClusterOperator resource to express their health and overall state.
+
+In order to address the existing plain provisioner limitation, this proposal also recommends expanding the plain+v0 bundle format, and enabling users to define health checks through a dedicated metadata file. A valid metadata file may be named "bundle.yaml" so clients can know where to find this file.
+
+For phase 0, the registry+v1 provisioner would be responsible for automatically generating this metadata file for PO packages. Future phases may introduce support for defining PO packages using OLM's plain bundle format, which will require operator authors to explicitly define this metadata file themselves.
+
+These metadata-level health checks can be expressed using JSONPath templates to declaratively define custom probe logic. When the plain provisioner is processing unpacked Bundle contents, and a manifests file has been defined, it’s responsible for serializing the values defined in the metadata file, and evaluating any defined custom probes. When an an individual probe evaluation fails, that state will be reflected in the BundleDeployment’s status.
+
+In the case that a probe was misconfigured, an admin may elect to toggle this behavior off entirely. The ProvisionerClass API, which doesn’t exist today, would be a candidate resource for providing this escape hatch behavior. Without that API in place today, the `plain.core.rukpak.io/enable-probes` BundleDeployment annotation key may serve as a short term solution. The plain provisioner implementation would be responsible for respecting that annotation key.
+
+Injecting this annotation may require changes to the top-level PlatformOperators resource to introduce a toggle field to control this behavior. When the PO manager sees this toggle has been disabled, it's responsible for propagating this configuration as an annotation to the set of BundleDeployment resources it manages.
+
+When reconciling a PO custom resource, the PO manager may need to propagate the underlying BundleDeployment's status back to the PO custom resource that's currently being processed. A separate controller may be responsible for watching, and listing the PO custom resource, and relaying that individual PO status information back to the aggregate ClusterOperator that controller is managing.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+Installer->>PO custom resource: Generates
```

Comment:
not in phase 0, right? but yes, potentially in the future (when the install-config has an api field for indicating which POs you want)

in phase 0 the user would create the PO resources and add them to the installer-generated manifests.

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/e6f356ef53203a97ab98f654704310f1e8a2f790.